traits/editor_factories.py

```diff
@@ -141,6 +150,30 @@ def date_editor():
     return DateEditor
 
 
+def datetime_editor():
+    """ Factory function that returns an editor with date & time for
+    editing DateTime values.
+    """
+    global DateTimeEditor
+
+    if DateTimeEditor is None:
+        try:
+            from traitsui.api import DateTimeEditor
+
+            DateTimeEditor = DateTimeEditor()
+
+        except ImportError:
+
+            logger.warn(msg="Could not import DateTimeEditor from "
+                            "traitsui.api, using TextEditor instead")
+
+            from traitsui.api import TextEditor
```

Comment:
The problem is that the TextEditor is actually useless here, since it cannot output datetime.datetime required by the DateTime trait. The trait type constraint will not be satisfied.

Reply:
The `TextEditor` can be given an `evaluator` argument that is a function that can be used to convert the raw text of the text field to a `datetime.datetime`.  A similar `format_func` argument can be used for the reverse conversion.

I can think of two reasonable options:

- use `isoformat` and `fromisoformat` methods, but `fromisoformat` is only available for Python >= 3.7, so we'd need to provide a replacement for 3.5 and 3.6
- use `strftime` and `strptime` methods; in this case there is a question of what format string to use. Something which matches ISO conventions is probably simplest, the alternative would be to mess around with locales or pybabel, which seems too much complexity for a fall-back.